### PR TITLE
[Feature] Add docker compose and more dockerfiles

### PR DESCRIPTION
**Pull request by wangruin**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: docker/docker-compose/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# docker-compose
+This is a docker compose instance for starrocks. The repo used for you learn have to use and test starrocks. Deploys starrocks and start one fe instance, three be instances.
```
Comment (kevincai) [inline docker/docker-compose/README.md:2]:
```suggestion
This is a docker compose instance for starrocks which is a good start to learn how to test and use starrocks. The compose deploys starrocks with one fe instance and three be instances by default.
```

File: docker/dockerfiles/Dockerfile_be_centos [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM centos:7
+
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+
+RUN yum install -y tzdata openssl curl vim ca-certificates fontconfig gzip tar mysql java-11-openjdk
+
+COPY /be /opt/starrocks/be/
+Copy be_entrypoint.sh be_prestop.sh /opt/starrocks/
```
Comment (kevincai) [inline docker/dockerfiles/Dockerfile_be_centos:8]: these scripts are not available in starrocks repo. It is a little weird to check in a partial done works into starrocks repo.

**Review by kevincai — COMMENTED** [review]
suggest refine the content of `docker/docker-compose/README.md`.

File: docker/docker-compose/README.md (hunk 1 above)
Comment (wangruin) [reply docker/docker-compose/README.md:2]: ok,thanks

File: docker/dockerfiles/Dockerfile_be_centos [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM centos:7
+
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+
+RUN yum install -y tzdata openssl curl vim ca-certificates fontconfig gzip tar mysql java-11-openjdk
+
+COPY ../../output/be /opt/starrocks/be/
+Copy ../bin/be_entrypoint.sh ../bin/be_prestop.sh /opt/starrocks/
```
Comment (kevincai) [inline docker/dockerfiles/Dockerfile_be_centos:8]: if docker build is running with `./dockerfiles` as build root, it is not possible to reference ../bin/ in dockerfile context

Comment (kevincai) [reply docker/dockerfiles/Dockerfile_be_centos:8]: please double check this file path.

**Review by kevincai — DISMISSED** [review]

**Comment by wangruin** [comment]
run starrocks_admit_test

File: docker/docker-compose/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    hostname: starrocks-be1
+    container_name: starrocks-be1
+    depends_on:
+      - "starrocks-fe"
+    volumes:
+      - ../../conf/be.conf:/opt/starrocks/be/conf/be.conf
+      #- ./data/starrocks-be1/storage:/opt/starrocks/be/storage
+```
+
+The comments in the above example defines the mount paths and volumes that FE and BE can acceess. You need to use the root user to deploy StarRocks with Docker. Therefore, you must delete the comments in the example if you use mount volumes to deploy StarRocks with mount volumes.
```
Comment (ss892714028) [inline docker/docker-compose/README.md:60]:
```suggestion
The commented-out sections in the above example YAML file define the mount paths and volumes that FE and BE use to persist data. Note that root privilege is required to deploy StarRocks with Docker with persistent volume. 
```

File: docker/docker-compose/README.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+## Troubleshooting
+
+When you connect to the cluster, StarRocks may return the following error:
+
+```shell
+ ERROR 2003 (HY000): Can't connect to MySQL server on 'starrocks-fe:9030' (111)
+```
+
+The reason may be that the BE node was started earlier than that the FE node was started. To solve this problem, you can re-run the `docker compose up` command, or manually add the BE node to the cluster using the following command:
```
Comment (ss892714028) [inline docker/docker-compose/README.md:88]:
```suggestion
The reason may be that the BE node was started before the FE node is ready. To solve this problem, re-run the docker compose up command, or manually add the BE node to the cluster using the following command:
```

**Comment by wangruin** [comment]
run starrocks_admit_test

**Review by shileifu — APPROVED** [review]

**Review by ss892714028 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by kevincai — APPROVED** [review]

**Review by shileifu — APPROVED** [review]

**Comment by shileifu** [comment]
+1

**Review by shileifu — APPROVED** [review]
+1

**Review by shileifu — APPROVED** [review]

**Review by shileifu — APPROVED** [review]

**Pull request merged**